core/crypto/aes/doc.odin

```diff
@@ -0,0 +1,47 @@
+/*
+package aes implements the AES block cipher and some common modes.
+
+
+An example of how to use 'AES-ECB instance' to decrypt binary files
+
+```odin
+
+package foo
+
+import "core:fmt"
+import "core:crypto/aes"
+import "core:os"
+import "core:bytes"
+
+main :: proc() {
+    data, ok := os.read_entire_file("data.bin")
+    if !ok {fmt.eprintln("Error Reading File")}
+
+    key := transmute([]u8)string("aeskey")
+    ctx: aes.Context_ECB
+
+    aes.init_ecb(&ctx, key)
+
+    if len(data) % aes.BLOCK_SIZE != 0 {
```

Comment:
Perhaps move this block up, underneath the check for properly reading the file. We're initializing for no reason if we return when this check fails.